### DISCUSSION TITLE
MQTT Publisher and Home Assistant

### DISCUSSION THREAD

**Discussion by redstormsju** [netalertx/NetAlertX#648] [Q&A]
Hello all, I’ve successfully integrated NetAlert X MQTT publisher. I’m running MQTT in a docker container where I also run Home Assistant.  In my Home Assistant MQTT integration, I see all of the devices that NetAlertX has picked up on my network. 

I’ve set up the MQTT schedule in NetAlert X to run at a specific time frame…several minutes after the arp scan. Ion the attached picture you can see the device as well as when the last update came through the mqtt publisher. If you see, there is a discrepancy between devices.  Why would some devices show more recent updates than others despite my mqtt schedule is set to a very specific time that is more frequent than reflected in the picture?  I’ve seen some reflect 7 hours ago.  My schedule is set to every 15th minute of every hour. 
![IMG_0490](https://github.com/jokob-sk/NetAlertX/assets/102559421/90e0452c-edea-4751-97b4-3859c3c1775d)

**Comment by jokob-sk** [comment]
Hey @redstormsju ,

If I remember correctly then only devices with a change will be updated/pushed to MQTT. So that's why you might see a discrepancy in the last update time. 

You can also check the log file to make sure the script isn't prematurely aborted and if so, increase the timeout from 10 to e.g. 30s per device. 

Hope this helps,
j

**Reply by redstormsju** [reply]
That makes sense. From what I see in the logs, the Mqtt push seems to be running smoothly and I don't see any errors. I am not versed in this but am confident based on the logs, there are no errors being reported so your feedback makes sense. Thank you!
